flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -451,31 +452,49 @@ class PFTBuilder {
   void rewriteIfGotos() {
     using T = struct {
       lower::pft::EvaluationList::iterator ifConstructIt;
-      parser::Label ifTargetLabel;
+      parser::Label ifTargetLabel; // 0 for cycle statements
```

Comment:
Instead of a special value, you might want to consider a bool flag here to make the distinction. e.g., would EXIT have another special value, etc.?